moved/src/move_execution/tests/utils.rs

```diff
@@ -0,0 +1,631 @@
+    pub genesis_config: GenesisConfig,
+    /// Transaction signer
+    pub signer: Signer,
+    /// Move address for contract deployment
+    pub move_address: AccountAddress,
+}
+
+impl TestContext {
+    /// Creates a new test context with initialized state and default signer
+    pub fn new() -> Self {
+        let genesis_config = GenesisConfig::default();
+        let mut state = InMemoryState::new();
+        init_state(&genesis_config, &mut state);
+
+        Self {
+            state,
+            genesis_config,
+            signer: Signer::new(&PRIVATE_KEY),
+            move_address: EVM_ADDRESS.to_move_address(),
+        }
+    }
+
+    /// Deploys a Move contract module and returns its ModuleId
+    ///
+    /// # Arguments
+    /// * `module_name` - Name of the module to deploy
+    ///
+    /// # Returns
+    /// The ModuleId of the deployed contract
+    pub fn deploy_contract(&mut self, module_name: &str) -> ModuleId {
+        let module_bytes = self.compile_module(module_name, self.move_address);
+        let (tx_hash, tx) = create_transaction(&mut self.signer, TxKind::Create, module_bytes);
+        let transaction = TestTransaction::new(tx, tx_hash);
+        let outcome = self.execute_tx(&transaction).unwrap();
+        self.state.apply(outcome.changes).unwrap();
+
+        let module_id = ModuleId::new(self.move_address, Identifier::new(module_name).unwrap());
+        assert!(
+            self.state
+                .resolver()
+                .get_module(&module_id)
+                .unwrap()
+                .is_some(),
+            "Code should be deployed"
+        );
+        module_id
+    }
+
+    /// Executes a Move script with arguments
+    ///
+    /// # Arguments
+    /// * `script_name` - Name of the script to execute
+    /// * `local_deps` - Local module dependencies needed by the script
+    /// * `args` - Arguments to pass to the script
+    pub fn run_script(
+        &mut self,
+        script_name: &str,
+        local_deps: &[&str],
+        args: Vec<TransactionArgument>,
+    ) -> Vec<Log<LogData>> {
+        let script_bytes = self.compile_script(script_name, local_deps, args);
+        let (tx_hash, tx) = create_transaction(&mut self.signer, TxKind::Create, script_bytes);
+        let transaction = TestTransaction::new(tx, tx_hash);
+        let outcome = self.execute_tx(&transaction).unwrap();
+        self.state.apply(outcome.changes).unwrap();
+        // Script transaction should succeed
+        outcome.vm_outcome.unwrap();
+        outcome.logs
+    }
+
+    /// Transfers ETH to a specified address with L1 cost considerations
+    ///
+    /// # Arguments
+    /// * `to` - Destination address for the transfer
+    /// * `amount` - Amount of ETH to transfer
+    /// * `l1_cost` - L1 cost associated with the transaction
+    /// * `expect_error` - Whether the transfer is expected to fail
+    pub fn transfer(&mut self, to: Address, amount: u64, l1_cost: u64, expect_error: bool) {
```

Comment:
Nit: passing a Boolean argument for internal control flow doesn't end up looking very intuitive at the call site. A better way to approach this is probably to return `outcome.vm_outcome` from this function so that the caller can decide to call `unwrap` or `unwap_err` as needed.